src/sync.js

```diff
@@ -0,0 +1,165 @@
+/* exported loadFromKinto */
+/* exported saveToKinto */
+
+let syncDebounce = null;
+
+
+const cryptographer = new Jose.WebCryptographer();
+cryptographer.setKeyEncryptionAlgorithm('A256KW');
+cryptographer.setContentEncryptionAlgorithm('A256GCM');
+
+function shared_key(key) {
+  return crypto.subtle.importKey(
+    'jwk',
+    { kty: key.kty, k: key.k.replace(/=/, '') },
+    'AES-KW',
+    true,
+    ['wrapKey', 'unwrapKey']
+  );
+}
+
+function encrypt(key, content) {
+  const encrypter = new JoseJWE.Encrypter(cryptographer, shared_key(key));
+  return encrypter.encrypt(JSON.stringify(content));
+}
+
+function decrypt(key, encrypted) {
+  const decrypter = new JoseJWE.Decrypter(cryptographer, shared_key(key));
+  return decrypter.decrypt(encrypted).then(result => {
+    return JSON.parse(result);
+  });
+}
+
+
+function loadFromKinto(client) {
+  // Get credentials and lastmodified
+  browser.storage.local.get(['credentials', 'contentWasSynced', 'last_modified'])
+    .then((data) => {
+      // XXX: Ask for an refresh token
+      // Query Kinto with the Bearer Token
+      if (!data.hasOwnProperty('credentials')) return;
+
+      client
+        .bucket('default')
+        .collection('notes')
+        .getRecord('singleNote', {
+          headers: { Authorization: `Bearer ${data.credentials.access_token}` }
+        })
+        .then(result => {
+          if (!data.hasOwnProperty('last_modified') ||
+              result.data.last_modified > data.last_modified) {
+            // If there is something in Kinto send unencrypted content to the sidebar
+            return decrypt(data.credentials.key, result['data']['content'])
+              .then(content => {
+                browser.runtime.sendMessage({
+                  action: 'kinto-loaded',
+                  data: content,
+                  contentWasSynced: data.contentWasSynced,
+                  last_modified: result.data.last_modified
+                });
+              })
+              .catch(() => {
+                // In case we cannot decrypt the message
+                if (data.credentials.key.kid < result.data.kid) {
+                  // If the key date is greater than current one, log the user out.
+                  return browser.storage.local.remove('credentials');
+                } else {
+                  // If the key date is older than the current one, we can't help
+                  // because there is no way we get the previous key.
+                  // Flush the server ALA sync
+                  return client
+                    .bucket('default')
+                    .collection('notes')
+                    .deleteRecord('singleNote', {
+                      headers: { Authorization: `Bearer ${data.credentials.access_token}` }
+                    });
+                }
+              });
+          }
+        })
+        .catch(error => {
+          if (/HTTP 404/.test(error.message)) {
```

Comment:
Not sure if we want some more friendly wrappers for these, such as `is404()`, `is401() {}`, `is412() {}`, etc.

or maybe something even more generic:

```js
function checkStatus(msg, code) {
  return new RegExp(`HTTP ${code}`).test(msg);
}

if (checkStatus(error.message, 404)) {
  ...
}

// function is404(msg) {
//  return checkStatus(msg, 404);
// }
```